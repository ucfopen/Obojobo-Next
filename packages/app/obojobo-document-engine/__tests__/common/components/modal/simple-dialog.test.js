import { mount } from 'enzyme'
import React from 'react'

jest.mock('../../../../src/scripts/common/util/modal-util')

import SimpleDialog from '../../../../src/scripts/common/components/modal/simple-dialog'
import ModalUtil from '../../../../src/scripts/common/util/modal-util'
import renderer from 'react-test-renderer'

describe('SimpleDialog', () => {
	test('SimpleDialog ok', () => {
		const component = renderer.create(
			<SimpleDialog ok onCancel={jest.fn()} onConfirm={jest.fn()}>
				Content
			</SimpleDialog>
		)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('SimpleDialog noOrYes', () => {
		const component = renderer.create(
			<SimpleDialog noOrYes onCancel={jest.fn()} onConfirm={jest.fn()}>
				Content
			</SimpleDialog>
		)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('SimpleDialog yesOrNo', () => {
		const component = renderer.create(
			<SimpleDialog yesOrNo onCancel={jest.fn()} onConfirm={jest.fn()}>
				Content
			</SimpleDialog>
		)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('SimpleDialog cancelOk', () => {
		const component = renderer.create(
			<SimpleDialog cancelOk onCancel={jest.fn()} onConfirm={jest.fn()}>
				Content
			</SimpleDialog>
		)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('SimpleDialog no type', () => {
		const component = renderer.create(
			<SimpleDialog onCancel={jest.fn()} onConfirm={jest.fn()}>
				Content
			</SimpleDialog>
		)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('Other keys dont close modal', () => {
		const onConfirm = jest.fn()
		const component = mount(
			<SimpleDialog onCancel={jest.fn()} onConfirm={onConfirm}>
				Content
			</SimpleDialog>
		)

		expect(onConfirm).toHaveBeenCalledTimes(0)

		document.dispatchEvent(new KeyboardEvent('keyup', { key: 'r' }))

		expect(onConfirm).toHaveBeenCalledTimes(0)

		component.unmount()
	})

	test('SimpleDialog ok click', () => {
		const onCancel = jest.fn()
		const onConfirm = jest.fn()

		const component = mount(
			<SimpleDialog ok onCancel={onCancel} onConfirm={onConfirm}>
				Content
			</SimpleDialog>
		)

		const leftButton = component.find('button').at(1)

		leftButton.simulate('click')

		expect(onCancel).toHaveBeenCalledTimes(0)
		expect(onConfirm).toHaveBeenCalledTimes(1)
	})

	test('SimpleDialog noOrYes click', () => {
		const onCancel = jest.fn()
		const onConfirm = jest.fn()

		const component = mount(
			<SimpleDialog noOrYes onCancel={onCancel} onConfirm={onConfirm}>
				Content
			</SimpleDialog>
		)

		const leftButton = component.find('button').at(1)
		const rightButton = component.find('button').at(2)

		leftButton.simulate('click')

		expect(onCancel).toHaveBeenCalledTimes(1)
		expect(onConfirm).toHaveBeenCalledTimes(0)

		rightButton.simulate('click')

		expect(onCancel).toHaveBeenCalledTimes(1)
		expect(onConfirm).toHaveBeenCalledTimes(1)
	})

	test('SimpleDialog yesOrNo click', () => {
		const onCancel = jest.fn()
		const onConfirm = jest.fn()

		const component = mount(
			<SimpleDialog yesOrNo onCancel={onCancel} onConfirm={onConfirm}>
				Content
			</SimpleDialog>
		)

		const leftButton = component.find('button').at(1)
		const rightButton = component.find('button').at(2)

		leftButton.simulate('click')

		expect(onCancel).toHaveBeenCalledTimes(0)
		expect(onConfirm).toHaveBeenCalledTimes(1)

		rightButton.simulate('click')

		expect(onCancel).toHaveBeenCalledTimes(1)
		expect(onConfirm).toHaveBeenCalledTimes(1)
	})

	test('SimpleDialog cancelOk click', () => {
		const onCancel = jest.fn()
		const onConfirm = jest.fn()

		const component = mount(
			<SimpleDialog cancelOk onCancel={onCancel} onConfirm={onConfirm}>
				Content
			</SimpleDialog>
		)

		const leftButton = component.find('button').at(1)
		const rightButton = component.find('button').at(2)

		leftButton.simulate('click')

		expect(onCancel).toHaveBeenCalledTimes(1)
		expect(onConfirm).toHaveBeenCalledTimes(0)

		rightButton.simulate('click')

		expect(onCancel).toHaveBeenCalledTimes(1)
		expect(onConfirm).toHaveBeenCalledTimes(1)
	})

	test('SimpleDialog cancelOk click defaults', () => {
		const component = mount(<SimpleDialog cancelOk>Content</SimpleDialog>)

		const leftButton = component.find('button').at(1)
		const rightButton = component.find('button').at(2)

		leftButton.simulate('click')

		expect(ModalUtil.hide).toHaveBeenCalledTimes(1)

		rightButton.simulate('click')

		expect(ModalUtil.hide).toHaveBeenCalledTimes(2)
	})

	test('focusOnFirstElement with props', () => {
		const focusFirst = jest.fn()
		const component = mount(<SimpleDialog focusOnFirstElement={focusFirst}>Content</SimpleDialog>)

		component.instance().focusOnFirstElement()

		expect(focusFirst).toHaveBeenCalled()
	})

	test('focusOnFirstElement without props', () => {
		const component = mount(<SimpleDialog cancelOk>Content</SimpleDialog>)

		component.instance().focusOnFirstElement()

		expect(component.html()).toMatchSnapshot()
	})
})
