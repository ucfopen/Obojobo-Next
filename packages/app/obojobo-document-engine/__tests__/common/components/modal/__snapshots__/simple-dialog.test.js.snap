// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleDialog SimpleDialog cancelOk 1`] = `
<div
  className="obojobo-draft--components--modal--simple-dialog"
>
  <div
    className="obojobo-draft--components--modal--dialog "
    style={null}
  >
    <div
      aria-labelledby="obojobo-draft--components--modal--modal--content"
      className="obojobo-draft--components--modal--modal"
      role="dialog"
    >
      <input
        className="first-tab"
        onFocus={[Function]}
        type="text"
      />
      <div
        className="obojobo-draft--components--delete-button"
      >
        <button
          onClick={[MockFunction]}
        >
          Delete
        </button>
      </div>
      <div
        className="content"
        id="obojobo-draft--components--modal--modal--content"
      >
        <div
          className="dialog-content"
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          Content
        </div>
        <div
          className="controls"
        >
          <div
            className="obojobo-draft--components--button alt-action is-not-dangerous align-center"
            contentEditable={false}
          >
            <button
              className="button"
              contentEditable={false}
              onClick={[MockFunction]}
            >
              Cancel
            </button>
          </div>
          <div
            className="obojobo-draft--components--button is-not-dangerous align-center"
            contentEditable={false}
          >
            <button
              className="button"
              contentEditable={false}
              onClick={[MockFunction]}
            >
              OK
            </button>
          </div>
        </div>
      </div>
      <input
        className="last-tab"
        onFocus={[Function]}
        type="text"
      />
    </div>
  </div>
</div>
`;

exports[`SimpleDialog SimpleDialog no type 1`] = `
<div
  className="obojobo-draft--components--modal--simple-dialog"
>
  <div
    className="obojobo-draft--components--modal--dialog "
    style={null}
  >
    <div
      aria-labelledby="obojobo-draft--components--modal--modal--content"
      className="obojobo-draft--components--modal--modal"
      role="dialog"
    >
      <input
        className="first-tab"
        onFocus={[Function]}
        type="text"
      />
      <div
        className="obojobo-draft--components--delete-button"
      >
        <button
          onClick={[MockFunction]}
        >
          Delete
        </button>
      </div>
      <div
        className="content"
        id="obojobo-draft--components--modal--modal--content"
      >
        <div
          className="dialog-content"
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          Content
        </div>
        <div
          className="controls"
        >
          <div
            className="obojobo-draft--components--button alt-action is-not-dangerous align-center"
            contentEditable={false}
          >
            <button
              className="button"
              contentEditable={false}
              onClick={[MockFunction]}
            >
              Cancel
            </button>
          </div>
          <div
            className="obojobo-draft--components--button is-not-dangerous align-center"
            contentEditable={false}
          >
            <button
              className="button"
              contentEditable={false}
              onClick={[MockFunction]}
            >
              OK
            </button>
          </div>
        </div>
      </div>
      <input
        className="last-tab"
        onFocus={[Function]}
        type="text"
      />
    </div>
  </div>
</div>
`;

exports[`SimpleDialog SimpleDialog noOrYes 1`] = `
<div
  className="obojobo-draft--components--modal--simple-dialog"
>
  <div
    className="obojobo-draft--components--modal--dialog "
    style={null}
  >
    <div
      aria-labelledby="obojobo-draft--components--modal--modal--content"
      className="obojobo-draft--components--modal--modal"
      role="dialog"
    >
      <input
        className="first-tab"
        onFocus={[Function]}
        type="text"
      />
      <div
        className="obojobo-draft--components--delete-button"
      >
        <button
          onClick={[MockFunction]}
        >
          Delete
        </button>
      </div>
      <div
        className="content"
        id="obojobo-draft--components--modal--modal--content"
      >
        <div
          className="dialog-content"
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          Content
        </div>
        <div
          className="controls"
        >
          <div
            className="obojobo-draft--components--button is-not-dangerous align-center"
            contentEditable={false}
          >
            <button
              className="button"
              contentEditable={false}
              onClick={[MockFunction]}
            >
              No
            </button>
          </div>
          <span
            className="text"
          >
            or
          </span>
          <div
            className="obojobo-draft--components--button is-not-dangerous align-center"
            contentEditable={false}
          >
            <button
              className="button"
              contentEditable={false}
              onClick={[MockFunction]}
            >
              Yes
            </button>
          </div>
        </div>
      </div>
      <input
        className="last-tab"
        onFocus={[Function]}
        type="text"
      />
    </div>
  </div>
</div>
`;

exports[`SimpleDialog SimpleDialog ok 1`] = `
<div
  className="obojobo-draft--components--modal--simple-dialog"
>
  <div
    className="obojobo-draft--components--modal--dialog "
    style={null}
  >
    <div
      aria-labelledby="obojobo-draft--components--modal--modal--content"
      className="obojobo-draft--components--modal--modal"
      role="dialog"
    >
      <input
        className="first-tab"
        onFocus={[Function]}
        type="text"
      />
      <div
        className="obojobo-draft--components--delete-button"
      >
        <button
          onClick={[MockFunction]}
        >
          Delete
        </button>
      </div>
      <div
        className="content"
        id="obojobo-draft--components--modal--modal--content"
      >
        <div
          className="dialog-content"
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          Content
        </div>
        <div
          className="controls"
        >
          <div
            className="obojobo-draft--components--button is-not-dangerous align-center"
            contentEditable={false}
          >
            <button
              className="button"
              contentEditable={false}
              onClick={[MockFunction]}
            >
              OK
            </button>
          </div>
        </div>
      </div>
      <input
        className="last-tab"
        onFocus={[Function]}
        type="text"
      />
    </div>
  </div>
</div>
`;

exports[`SimpleDialog SimpleDialog yesOrNo 1`] = `
<div
  className="obojobo-draft--components--modal--simple-dialog"
>
  <div
    className="obojobo-draft--components--modal--dialog "
    style={null}
  >
    <div
      aria-labelledby="obojobo-draft--components--modal--modal--content"
      className="obojobo-draft--components--modal--modal"
      role="dialog"
    >
      <input
        className="first-tab"
        onFocus={[Function]}
        type="text"
      />
      <div
        className="obojobo-draft--components--delete-button"
      >
        <button
          onClick={[MockFunction]}
        >
          Delete
        </button>
      </div>
      <div
        className="content"
        id="obojobo-draft--components--modal--modal--content"
      >
        <div
          className="dialog-content"
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          Content
        </div>
        <div
          className="controls"
        >
          <div
            className="obojobo-draft--components--button is-not-dangerous align-center"
            contentEditable={false}
          >
            <button
              className="button"
              contentEditable={false}
              onClick={[MockFunction]}
            >
              Yes
            </button>
          </div>
          <span
            className="text"
          >
            or
          </span>
          <div
            className="obojobo-draft--components--button is-not-dangerous align-center"
            contentEditable={false}
          >
            <button
              className="button"
              contentEditable={false}
              onClick={[MockFunction]}
            >
              No
            </button>
          </div>
        </div>
      </div>
      <input
        className="last-tab"
        onFocus={[Function]}
        type="text"
      />
    </div>
  </div>
</div>
`;

exports[`SimpleDialog focusOnFirstElement without props 1`] = `"<div class=\\"obojobo-draft--components--modal--simple-dialog\\"><div class=\\"obojobo-draft--components--modal--dialog \\"><div class=\\"obojobo-draft--components--modal--modal\\" role=\\"dialog\\" aria-labelledby=\\"obojobo-draft--components--modal--modal--content\\"><input class=\\"first-tab\\" type=\\"text\\"><div class=\\"obojobo-draft--components--delete-button\\"><button>Delete</button></div><div class=\\"content\\" id=\\"obojobo-draft--components--modal--modal--content\\"><div class=\\"dialog-content\\" style=\\"text-align: center;\\">Content</div><div class=\\"controls\\"><div class=\\"obojobo-draft--components--button alt-action is-not-dangerous align-center\\" contenteditable=\\"false\\"><button class=\\"button\\" contenteditable=\\"false\\">Cancel</button></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\" contenteditable=\\"false\\"><button class=\\"button\\" contenteditable=\\"false\\" tabindex=\\"0\\">OK</button></div></div></div><input class=\\"last-tab\\" type=\\"text\\"></div></div></div>"`;
